util-images/network/netperfbenchmark/pkg/worker/worker.go

```diff
@@ -0,0 +1,458 @@
+		klog.Error("unable to obtain Stdout:", err)
+		resultStatus <- err.Error()
+		return
+	}
+	eout, err := cmd.StderrPipe()
+	if err != nil {
+		klog.Error("unable to obtain Stderr:", err)
+		resultStatus <- err.Error()
+		return
+	}
+	multiRdr := io.MultiReader(out, eout)
+	err = cmd.Start()
+	if err != nil {
+		resultStatus <- err.Error()
+	}
+	scanOutput(&multiRdr)
+}
+
+func scanOutput(out *io.Reader) {
+	scanner := bufio.NewScanner(*out)
+	klog.Info("Starting scan for output")
+	for scanner.Scan() {
+		l := scanner.Text()
+		// klog.Info(line)
+		if l == "" {
+			continue
+		}
+		result = append(result, l)
+	}
+	klog.Info("Command executed,sending result back")
+	if err := scanner.Err(); err != nil {
+		klog.Error("Error", err)
+		resultStatus <- err.Error()
+		return
+	}
+	resultStatus <- "OK"
+}
+
+func parseResult() ([]float64, error) {
+	klog.Info("Parsing", result[0])
+	switch tstProtocol {
+	case protocolTCP:
+		return parseTCP(result), nil
+	case protocolUDP:
+		return parseUDP(result), nil
+	case protocolHTTP:
+		return parseHTTP(result), nil
+	default:
+		return nil, errors.New("invalid protocol:" + result[0])
+
+	}
+}
+
+func parseTCP(result []string) []float64 {
+	klog.Info("In parseTCP")
+	sumResult := make([]float64, 0, tcpMtrCnt)
+	unitReg := regexp.MustCompile(`\[\s+|\]\s+|KBytes\s+|KBytes/sec\s*|sec\s+|ms\s+|us\s*`)
+	mulSpaceReg := regexp.MustCompile(`\s+`)
+	hypSpcReg := regexp.MustCompile(`\-\s+`)
+	cnt := 0
+	sessionID := make(map[string]bool)
+	for _, op := range result {
+		klog.Info(op)
+		frmtString := hypSpcReg.ReplaceAllString(mulSpaceReg.ReplaceAllString(unitReg.ReplaceAllString(op, " "), " "), "-")
+		if !strings.Contains(frmtString, "0.0-"+tstDuration+".0") { //single digit dur has probs
+			continue
+		}
+		klog.Info("Trim info:", frmtString)
+		split := strings.Split(frmtString, " ")
+		//for bug in iperf tcp
+		//if the record is for the complete duration of run
+		if len(split) >= 3 && "SUM" != split[1] && split[2] == "0.0-"+tstDuration+".0" {
+			if _, ok := sessionID[split[1]]; ok {
+				continue
+			}
+			for i, v := range split {
+				klog.Info("Split", i, ":", v)
+				if i == 1 {
+					sessionID[v] = true
+					continue
+				}
+				//first index and hte last is ""
+				if i == 0 || i == 2 || i == 5 {
+					continue
+				}
+				tmp, err := strconv.ParseFloat(v, 64)
+				if err != nil {
+					klog.Error("Conversion error", err)
+				}
+				if len(sumResult) <= 1 {
+					sumResult = append(sumResult, tmp)
+				} else {
+					switch iperfTCPFn[i-3] {
+					case "Sum":
+						sumResult[i-3] = tmp + sumResult[i-3]
+					case "Avg":
+						sumResult[i-3] = (float64(cnt)*tmp + sumResult[i-3]) / (float64(1 + cnt))
+					case "Min":
+						sumResult[i-3] = math.Min(tmp, sumResult[i-3])
+					case "Max":
+						sumResult[i-3] = math.Max(tmp, sumResult[i-3])
+					}
+				}
+				cnt++
+			}
+		}
+	}
+	klog.Info("Final output:", sumResult)
+	return sumResult
+}
+
+func parseUDP(result []string) []float64 {
```

Comment:
There is huge code duplication between those parse methods.

Can we please reuse that more?

Reply:
Modified the code for reusing logic.